src/ClientData/CaptureDataTest.cpp

```diff
@@ -46,6 +50,27 @@ constexpr std::array<uint64_t, kTimersForSecondId> kSortedDurationsForSecondId =
 
 constexpr uint64_t kLargeInteger = 10'000'000'000'000'000;
 
+const ThreadStateSliceInfo kSlice1 =
+    ThreadStateSliceInfo(1000, orbit_grpc_protos::ThreadStateSlice::kInterruptibleSleep, 50, 100,
+                         orbit_base::kInvalidThreadId, orbit_base::kInvalidProcessId,
+                         ThreadStateSliceInfo::WakeupReason::kNotApplicable);
+const ThreadStateSliceInfo kSlice2 =
+    ThreadStateSliceInfo(1000, orbit_grpc_protos::ThreadStateSlice::kRunnable, 101, 150, 4200, 420,
+                         ThreadStateSliceInfo::WakeupReason::kUnblocked);
```

Comment:
`begin_timestamp_ns` and `end_timestamp_ns` form a half open interval. That means when `kSlice` goes from 50 to 100, then `kSlice2` should start at `100`.

This also means that this check:

```cpp
EXPECT_THAT(capture_data_.FindThreadStateSliceInfoFromTimestamp(1000, 100), Optional(kSlice1));
```

should return `kSlice2`.

Reply:
You mean from 50 to 99 then slice 2 starts at 100?

Reply:
So you would put 50 as the begin timestamp and 100 as the end timestamp. But when the FindThre... method is called with timestamp 100 it should return the second slice because 100 is not part of the first slice anymore.